### ISSUE TITLE
serving/xrefs: add support for detailed traces [phab:D1516]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1930]
_Created by **schroederc** at 2017-04-12 23:32:33_:

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-04-13 00:23:24_:

There are enough of these it might be worth putting the common case into a helper:

```
func tracePrintf(ctx context.Context, msg string, args ...interface{}) {
   if t, ok := trace.FromContext(ctx); ok {
      t.LazyPrintf(msg, args...)
   }
}
```

which would collapse al but one of these cases, I believe.

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2017-04-13 00:23:24_

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-04-13 15:54:52_:

Add tracePrintf helper

**Comment by kythe** [comment]
_**schroederc** closed via commit 0a8bf564955f5594bda4edd0135947120711b04c at 2017-04-13 16:58:16_
